__tests__/app.test.js

```diff
@@ -293,4 +293,97 @@ describe("app", () => {
         });
     });
   });
+  describe("/api/articles/:article_id/comments - POST", () => {
+    test("status 200 - adds a comment to the comment table", () => {
+      const newComment = {
+        username: "lurker",
+        body: "first ever review"
+      };
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send(newComment)
+        .expect(200)
+        .then(() => {
+          return request(app).get("/api/articles/2/comments");
+        })
+        .then(({ body: { comments } }) => {
+          expect(comments[0].body).toBe("first ever review");
+        });
+    });
+
+    test("status 200 - returns the added comment", () => {
+      const newComment = {
+        username: "lurker",
+        body: "first ever review"
+      };
+
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send(newComment)
+        .expect(200)
+        .then(({ body: { comment } }) => {
+          expect(comment).toEqual(
+            expect.objectContaining({
+              body: "first ever review",
+              votes: 0,
+              author: "lurker",
+              article_id: 2,
+              created_at: expect.any(String)
+            })
+          );
+        });
+    });
+    test("status 404 - article not found", () => {
+      const newComment = {
+        username: "lurker",
+        body: "first ever review"
+      };
+      return request(app)
+        .post("/api/articles/20/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("data not found");
+        });
+    });
+    test("status 400 - invalid article_id", () => {
+      const newComment = {
+        username: "lurker",
+        body: "first ever review"
+      };
+      return request(app)
+        .post("/api/articles/a/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("invalid input");
+        });
+    });
+    test("status 400 - comment added invalid form", () => {
+      const newComment = {
+        u: "lurker",
+        b: "first ever review"
+      };
+      return request(app)
+        .post("/api/articles/a/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("invalid input");
+        });
+    });
+    test("status 400 - comment added invalid data types", () => {
+      const newComment = {
+        username: 9,
+        body: 234
+      };
+      return request(app)
+        .post("/api/articles/a/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("invalid input");
+        });
+    });
+  });
```

Comment:
Need status 404 user does not exist. newComment = {
        username: "lurkerrrrrrrrrrrrrrrrr",
        body: "first ever review"
      };